app/javascript/components/timeline-options/timeline-chart.jsx

```diff
@@ -0,0 +1,54 @@
+import React, { useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import { LineChart } from '@carbon/charts-react';
+import { timelineUiOptions } from './timeline-helper';
+
+const TimelineChart = ({ data, title, buildTableData }) => {
+  const chartRef = useRef(null);
+  const chartOnClick = ({ detail }) => buildTableData(detail.datum);
+
+  useEffect(() => {
+    chartRef.current.chart.services.events.addEventListener(
+      'scatter-click',
+      chartOnClick
+    );
+  }, [chartRef]);
+
+  // Unmount
+  useEffect(
+    () => () => {
+      if (chartRef.current) {
+        chartRef.current.chart.services.events.removeEventListener(
+          'scatter-click',
+          chartOnClick
+        );
+      }
+    },
+    []
+  );
+
+  const LengthWarning = data.length === 5000 ?
+    <label className="bx--label"> {__('*Only 5000 events shown. Limit date range to avoid \"missing\" events.')} </label>
+    : <></>;
```

Comment:
found few lint errors and have a suggestion.

```
const LengthWarning = (
    <div className="bx--label">
      {__('*Only 5000 events shown. Limit date range to avoid missing events.')}
    </div>
  );
```